gramps/gen/filters/rules/family/_motherhasnameof.py

```diff
@@ -47,5 +55,13 @@ class MotherHasNameOf(HasNameOf):
     name = _("Families with mother with the <name>")
     description = _("Matches families whose mother has a specified " "(partial) name")
     category = _("Mother filters")
-    base_class = HasNameOf
-    apply = mother_base
+
+    def apply_to_one(self, db: Database, family: Family) -> bool:  # type: ignore[override]
+        mother_handle = family.mother_handle
+        if mother_handle:
+            mother = db.get_person_from_handle(mother_handle)
+            if mother:
+                return super().apply_to_one(db, mother)
+            else:
+                return False
```

Comment:
```suggestion
```
Unnecessary else.

Reply:
I think it is more explicit, and "Explicit is better than implicit."